scala/src/main/scala/com/xebia/functional/xef/scala/prompt/PromptTemplate.scala

```diff
@@ -0,0 +1,22 @@
+package com.xebia.functional.xef.scala.prompt
+
+import com.xebia.functional.xef.scala.auto.*
+import io.circe.Decoder
+
+trait PromptTemplate[A] {
+
+  def chain(template: String): A
+
+  extension (a: A) {
+    def chain[B: ScalaSerialDescriptor: Decoder](template: A => String): B =
+      ai(prompt[B](template(a)))
```

Comment:
I'm not very sure if this should live here or not, but what do you think about adding a infix operator? Something like `>>` so we can do this:
```scala
PromptTemplate[Play](Play("The power of Zuluastral", "Modern Era")) >> synopsisTemplate >> reviewTemplate >> scoreTemplate
```
Is it possible? Does it make sense? If not, just forget it! 😄

Reply:
I think this syntactic sugar is possible yes but would need to try it. 
I'll take a note, will do it in a future PR if it does work.